libtsuba/include/katana/TxnContext.h

```diff
@@ -3,53 +3,73 @@
 
 #include <set>
 
+#include "katana/RDGManifest.h"
+#include "katana/URI.h"
 #include "katana/config.h"
 
 namespace katana {
 
 class KATANA_EXPORT TxnContext {
 public:
-  void InsertNodePropertyRead(std::string rdg_dir, std::string name) {
-    node_properties_read_.insert(ConcatRDGProperty(rdg_dir, name));
+  TxnContext() {}
+
+  explicit TxnContext(bool auto_commit) : auto_commit_(auto_commit) {}
```

Comment:
Could you add some documentation about this constructor, e.g.,
```
/// Create a transaction context that commits changes when the context is destroyed.
///
/// This is typically used when calling from transaction unaware code like tests.
```

and the one above it
```
/// Create a transaction context. Changes are only committed when Commit is called.
```

Reply:
Done. I set the `auto_commit_{true}` by default.